manticore/platforms/evm.py

```diff
@@ -21,10 +23,25 @@
 
 logger = logging.getLogger(__name__)
 
-#fixme make it global using this https://docs.python.org/3/library/configparser.html
-#and save it at the workspace so results are reproducible
-config = namedtuple("config", "out_of_gas")
-config.out_of_gas = None  # 0: default not enough gas, 1 default to always enough gas, 2: for on both
+#Gas behaivor configuration
+# When gas is concrete the gas checks and calculation are pretty straigth forward
+# Though Gas can became symbolic in normal bytecode execution for example at instructions
+# MSTORE, MSTORE8, EXP, ... and every instruction with internal operation restricted by gas
+# This configuration variable allows the user to control and perhaps relax the gas calculation
+# 0: gas is faithfully accounted and checked at instruction level. State may get forked in OOG/NoOOG
+# 1: gas is faithfully accounted and checked at basic blocks limits. State may get forked in OOG/NoOOG
+# 2: Concretize gas. If the fee to be consumed gets to be symbolic. Choose some potential values and fork on those.
+# 3: Try not to OOG. If it may be enough gas we ignore the OOG case. A constraint is added to assert the gas is enough and the OOG state is ignored.
+# 4: OOG soon. If it may NOT be enough gas we ignore the normal case. A constraint is added to assert the gas is NOT enough and the other state is ignored.
+# 99: Ignore gas. Do not account for it. Do not OOG.
```

Comment:
Can we have a mode in which we adjust the gas so it exactly accounts all the given tx (or conditions/constraints)?

Reply:
Like what if one do `require(gasNow() == 0x1234)` ?

Reply:
In that case `gasNow()` will return (for option 1) a symbolic value describing it. 
And the state will be forked in `=0x1234` and  `!=0x1234`.
If you want to ignore one of those states is more a task for general state selection/policy.